Sources/Plasma/CoreLib/plFileSystem.cpp

```diff
@@ -460,7 +469,36 @@ plFileName plFileSystem::GetUserDataPath()
 
         _userData = plFileName::Join(ST::string::from_wchar(path), plProduct::LongName());
 #else
-        _userData = plFileName::Join(getenv("HOME"), "." + plProduct::LongName());
+        ST::string dir;
+#  if HS_BUILD_FOR_APPLE
+        char home[256] = "";
```

Comment:
According to the man page:

> The path buffer that is passed to **sysdir_get_next_search_path_enumeration**() will be filled with a null-terminated string, possibly containing as many as PATH_MAX-1 characters.

```suggestion
        char home[PATH_MAX] = "";
```

Reply:
Building on the suggestion by @dgelessus 

```suggestion
        char home[PATH_MAX]{};
```